fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/rdf/impl/TypeRdfContext.java

```diff
@@ -58,71 +61,36 @@ public TypeRdfContext(final FedoraResource resource,
 
         //include rdf:type for primaryType, mixins, and their supertypes
         concatRdfTypes();
+        if (resource() instanceof NonRdfSource) {
+            // gather versionability info from the parent
+            if (resource().getNode().getParent().isNodeType("mix:versionable")) {
```

Comment:
_Both the description and the bitstream were versioned_ and _both still are_. You have yet to explain to me why we would want only one of the two to be versioned if the other is not. The fact that the JCR versioning functionality is engaged through the upper and not the lower node seems to me to be utterly incidental. If you would like to offer a PR in which it moves, I would be happy to merge it.